frontend/template.go

```diff
@@ -0,0 +1,11 @@
+package frontend
+
+import "github.com/Azure/dalec"
+
+func DefaultLibexecSubpath(s *dalec.Spec, k string) string {
```

Comment:
This doesn't look right.
The way we handle this for the other types (like libs) is always `<base>/<spec name>/<subpath>/<artifact name>`

As an example, see how `Libs` are handled.

Also at this time I'd like to avoid abstracting the behavior here.

Reply:
Let me double check a real system for common libexec usage, though.

Reply:
Yeah, I think that's the way to go, and if someone needs something directly in `/usr/libexec` or somewhere else they can use a symlink.

Reply:
What about when packages are renamed from upstream? For example, if we have the moby-buildx package, we want the binary to go into `/usr/libexec/docker/cli-plugins/docker-buildx` rather than `/usr/libexec/moby-buildx/cli-plugins/docker-buildx`. IMO we should let the user do the right thing without adding extra steps.

Reply:
> This doesn't look right. The way we handle this for the other types (like libs) is always `<base>/<spec name>/<subpath>/<artifact name>`

That seems pretty non-standard. Pretty much everything in my `/usr/libexec` directory is either in `/usr/libexec` directly, or nested under a single subdirectory bearing the main application's name. There's only a few files nested deeper than the application's name.

> If someone needs something directly in `/usr/libexec` or somewhere else they can use a symlink.

That would violate the standard referenced above: "Applications may use a single subdirectory under /usr/libexec". If we create a symlink, we have one application (docker) with two subdirectorys under `/usr/libexec` (`/usr/libexec/docker` and `/usr/libexec/moby-buildx`).

Reply:
Maybe a separate field to overwrite the package name is worthwhile in this case.
Or just a bool like `OmitAutoPackageName`

Reply:
done